### PR TITLE
Zypper correct exit code checking

### PR DESCRIPTION
**Pull request by mcalmer**
Checking exit code of zypper in the correct way and raising an exception with message
in case of an error

Adapt tests and improve existing tests.
Wrote new test for zypper result checking.
### Tests written?
- [X] Yes
- [ ] No

### CODE REVIEW

**Comment by mcalmer** [comment]
@isbm : please have a look.

File: salt/modules/zypper.py [hunk 1]
```diff
@@ -505,7 +547,9 @@ def del_repo(repo):
     for alias in repos_cfg.sections():
         if alias == repo:
             cmd = _zypper('-x', 'rr', '--loose-auth', '--loose-query', alias)
-            doc = dom.parseString(__salt__['cmd.run'](cmd, output_loglevel='trace'))
+            ret = __salt__['cmd.run_all'](cmd, output_loglevel='trace')
+
```
Comment (isbm) [inline salt/modules/zypper.py]: Is this empty line is needed?

Comment (mcalmer) [inline salt/modules/zypper.py]: I think it makes the code more readable.

File: salt/modules/zypper.py [hunk 2]
```diff
@@ -1018,7 +1043,9 @@ def clean_locks():
     if not os.path.exists("/etc/zypp/locks"):
         return out
 
-    doc = dom.parseString(__salt__['cmd.run'](_zypper('-x', 'cl'), output_loglevel='trace'))
+    ret = __salt__['cmd.run_all'](_zypper('-x', 'cl'), output_loglevel='trace')
+
```
Comment (isbm) [inline salt/modules/zypper.py]: An empty line maybe can go away?

**Comment by isbm** [comment]
Thanks @mcalmer :+1: LGTM.

**Comment by cachedout** [comment]
Thanks, @mcalmer

**Pull request merged**